tests/test_edit_script.rs

```diff
@@ -51,46 +51,40 @@ use common::check_files;
 
 #[test]
 fn test_both_arenas_empty() {
-    let ast_from = Arena::new();
+    let ast_from: Arena<String, _> = Arena::new();
     let ast_to = Arena::new();
-    // Generate mappings between ASTs.
     let matcher_config = MyersConfig::new();
     let store = matcher_config.match_trees(ast_from, ast_to);
     assert!(store.from.borrow().is_empty());
     assert!(store.to.borrow().is_empty());
-    // Generate an edit script.
-    let gen_config: Box<EditScriptGenerator<String>> = Box::new(Chawathe96Config::new());
-    let _edit_script_wrapped = gen_config.generate_script(&store);
+    Box::new(Chawathe96Config::new()).generate_script(&store)
+                                     .ok();
 }
 
 #[test]
 #[should_panic]
 fn test_from_arena_empty() {
     let ast_from = Arena::new();
     let ast_to = parse_file("tests/empty.calc", &get_lexer("grammars/calc.l"), &get_parser("grammars/calc.y")).unwrap();
-    // Generate mappings between ASTs.
     let matcher_config = MyersConfig::new();
     let store = matcher_config.match_trees(ast_from, ast_to);
     assert!(store.from.borrow().is_empty());
     assert!(store.to.borrow().is_empty());
-    // Generate an edit script.
-    let gen_config: Box<EditScriptGenerator<String>> = Box::new(Chawathe96Config::new());
-    let _edit_script_wrapped = gen_config.generate_script(&store); // Panic.
+    Box::new(Chawathe96Config::new()).generate_script(&store)
+                                     .ok(); // Panic.
```

Comment:
`ok` doesn't panic?

Reply:
It does panic, yep

Reply:
What's the return type of `generate_script`? `Result::ok()` says "Converts self into an Option<T>, consuming self, and discarding the error, if any."

Reply:
I'm not sure that I follow this. The return type of `generate_script()` is `EditScriptResult<T>` which is an alias for `Result<EditScript<T>, ArenaError>`. The `ok()` swallows the `ArenaError`, but it doesn't effect any `panic!()`s. If we replace lines 73 and 74 above with a call to the `panic!` macro, the results of the unit tests don't change. 

Have I missed something?

Reply:
I think we're partly agreeing, but I think we should remove the text that says `panic` because it's not clear why it says that (to me, at least).

Reply:
I think I just put them in to make clear which LOC causes the panic.